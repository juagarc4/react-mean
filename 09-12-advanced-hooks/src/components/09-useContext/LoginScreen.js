import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext)
  const newUser = { id: 1, name: 'Raul', email: 'email@rgc.com' }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(newUser)}>
        Login
      </button>
    </div>
  )
}
