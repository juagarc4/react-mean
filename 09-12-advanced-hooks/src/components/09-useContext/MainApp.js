import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'
export const MainApp = () => {
  const user = {
    id: 1,
    name: 'Raul Garcia',
    email: 'email@rgc.com',
  }
  return (
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  )
}
