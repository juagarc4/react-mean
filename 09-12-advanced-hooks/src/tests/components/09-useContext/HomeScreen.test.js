import React from 'react'
import { shallow, mount } from 'enzyme'
import { HomeScreen } from 'components/09-useContext/HomeScreen'
import { UserContext } from 'components/09-useContext/UserContext'

describe('Tests in component <HomeScreen/>', () => {
  const user = {
    name: 'Raul',
    email: 'test@email.com',
  }
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  )

  test('Should show the component properly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
