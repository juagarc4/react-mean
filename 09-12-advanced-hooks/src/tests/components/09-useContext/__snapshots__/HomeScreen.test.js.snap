// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component <HomeScreen/> Should show the component properly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Raul",
   "email": "test@email.com"
}
  </pre>
</div>
`;
